Dockerfile

```diff
@@ -60,17 +68,20 @@ ENV MENDER_CLIENT_VERSION=$mender_client_version
 
 # NOTE: we are assuming generic ARM board here, needs to be extended later
 
-ENV PATH "$PATH:/usr/local/go/bin"
+ENV PATH "$PATH:/usr/local/go/bin:/armv6-eabihf--glibc--stable-2018.11-1/bin"
 ENV GOPATH "/root/go"
 
 RUN go get -d github.com/mendersoftware/mender
 WORKDIR $GOPATH/src/github.com/mendersoftware/mender
 RUN git checkout $MENDER_CLIENT_VERSION
 
+ENV CC "arm-buildroot-linux-gnueabihf-gcc"
+
 RUN env CGO_ENABLED=1 \
     CGO_CFLAGS="-I${LIBLZMA_INSTALL_PATH}/include" \
     CGO_LDFLAGS="-L${LIBLZMA_INSTALL_PATH}/lib" \
     CC=arm-linux-gnueabihf-gcc \
+    CC=$CC \
     GOOS=linux \
     GOARCH=arm make build
```

Comment:
We should add the parameter `GOARM=5`, same way we cross-compiling in `mender-dist-packages`. Or at least v6, else Go could potentially decide to use v7 instructions

Reply:
@lluiscampos I assume that Go wouldn't manage to use v7 instructions as `arm-buildroot-linux-gnueabihf-gcc` is tuned for v6, so you can set GOARM to whatever value, but it will be ignored.

Reply:
gcc is only used for the C parts of the client, the Go native compiler is still used for the Go parts. So indeed `GOARM` should be set to 6 to match what gcc will produce.

Reply:
Yes, I think it is ignored as the linker will not support it anyway.

But I still think better to set it up for v6/v5 in case this triggers errors in future releases of Go or something.

Reply:
Ok, GOARM=6 added.

Reply:
:+1: